array_api_tests/test_elementwise_functions.py

```diff
@@ -112,20 +107,20 @@ def test_acosh(x):
     # to nan, which is already tested in the special cases.
     assert_exactly_equal(domain, codomain)
 
-@given(two_numeric_dtypes.flatmap(lambda i: two_array_scalars(*i)))
-def test_add(args):
-    x1, x2 = args
+@given(two_mutual_arrays(numeric_dtype_objects))
+def test_add(x1_and_x2):
+    x1, x2 = x1_and_x2
```

Comment:
This is not ideal, but thinking about it, I think it's the best option. The alternative with shared stuff would just make the strategies really messy. It's really too bad that Python dropped the `def f((x, y))` syntax that used to be allowed in Python 2.

Reply:
This works great, thanks!